eclipse-theia/theia
# repo: drop support for node 10

#### What it does


Related: https://github.com/eclipse-theia/theia/issues/8184

The following pull-request drops support for **Node v10** from the framework.

The pull-request updates the following:
- updates the node `engines` tag (so that the minimum and maximum node versions are respected when building).
- updates any documentation related to the node version.
- updates the CI to drop node 10 builds.

#### How to test


- verify that it is no longer possible to build with node 10
- verify that the CI build for node 10 is successfully removed

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
